src/nvim/eval.c

```diff
@@ -6234,20 +6235,23 @@ char *save_tv_as_string(typval_T *tv, ptrdiff_t *const len, bool endnl)
   // Pre-calculate the resulting length.
   list_T *list = tv->vval.v_list;
   TV_LIST_ITER_CONST(list, li, {
-    *len += (ptrdiff_t)strlen(tv_get_string(TV_LIST_ITEM_TV(li))) + 1;
+    *len += (ptrdiff_t)strlen(tv_get_string(TV_LIST_ITEM_TV(li))) + (crlf ? 2 : 1);
   });
 
   if (*len == 0) {
     return NULL;
   }
 
-  char *ret = xmalloc((size_t)(*len) + endnl);
+  char *ret = xmalloc((size_t)(*len) + (size_t)endnl * (crlf ? 2 : 1));
```

Comment:
Seems like a mistake that we were adding `endnl` (a boolean) here? Instead, it should be used as a ternary which evaluates to an int literal?

Reply:
I thought it was intentional. False evaluating to 0, etc.

Reply:
If we change `endnl` to a int, this line can remain as it was.

Reply:
> False evaluating to 0, etc.

maybe, but technically there are some places where we cast int to `bool`. Doesn't matter in this case but it seems unnecessarily clever. In any case, it's moot because it seems like a good idea to change the `endnl` to `int` instead of adding a new `crlf` param.

Reply:
I'm a little unclear on turning `endnl` into an int. Wrote my confusion in the comments below. Meanwhile, I committed your ternary op suggestion. You're right. At least before it made sense in that it shaved off two/three characters of code. In my case, it was just idiotic.